jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -85,10 +92,19 @@ public static boolean getDefaultUseCaches()
      * @return A Resource object.
      * @throws MalformedURLException Problem accessing URI
      */
-    public static Resource newResource(URI uri)
-        throws MalformedURLException
+    public static Resource newResource(URI uri) throws IOException
     {
-        return newResource(uri.toURL());
+        try
+        {
+            Path path = Paths.get(uri);
+            return newResource(path);
+        }
+        catch (FileSystemNotFoundException e)
+        {
+            FileSystem fileSystem = FileSystems.newFileSystem(uri, EMPTY_ENV);
```

Comment:
if this is a "jar:" or "jar:file", shouldn't we explicitly call constructor of `JarResource` or `JarFileResource` as those classes can internally use a filesystem pool (like what @joakime developed in his branch).

We still might not release it, but at least we would avoid creating it multiple times

Reply:
Oh - I see those classes have been deleted.     But we still need the concept of a FileSystem pool for jars, so perhaps that can be done in PathResource?  or a special subclass of it.    But we can't re-unpack a jar for every resource.

Reply:
Thinking about the problem, I think the proper solution is to have a `PathResource` for `file:` URIs as those don't need to be pooled and a `PoolingPathResource` for all other schemes that extends `PathResource` and does FS pooling with ref-counting like what @joakime did.